app/controllers/login_controller.rb

```diff
@@ -0,0 +1,35 @@
+class LoginController < ApplicationController
+
+	get "/login" do 
+    if logged_in?
+      redirect "/users/#{current_user.slug}"
+    else
+      @title = "Fitness Tracker - Sign In"
+      erb :login
+    end
+  end
+
+  post "/login" do
+    if !logged_in?
+      user = User.find_by(username: params[:username])
+      if user && user.authenticate(params[:password])
+        session[:id] = user.id

```

Comment:
I would think about calling the session key [:user_id] instead of [:id] because it is more concise, since this isn't really the id of the session it is the user id in the session.